### DISCUSSION TITLE
I accidentally deleted stuff, but don't understand why.

### DISCUSSION THREAD

**Discussion by snowmangoh** [advplyr/audiobookshelf#362] [General]
So I had a book who's folder structure looked like:
- AUTHOR
-- SERIES
--- BOOK
---- CD01
----- 01- TRACK.mp3
---- CD02... etc.

I renamed all of the tracks (like "CD01 - 01- Track.mp3"), moved them to the BOOK folder. I then went to the errors, and deleted everything (10 CDs/"covers" worth) that said the directory wasn't found. I then realized it had deleted all the way up to the Book folder. Is this expected behavior?

My only thought was since it had labeled all of the CD folders as part of a series called BOOK, that the logic said "He deleted the entire series, so I can delete that root series folder as well."

Not sure if this would fall under "bug" or "enhancement", or if I did something obviously stupid.

**Comment by advplyr** [comment]
Yes this is expected behavior but certainly not ideal behavior. Let me explain what I think happened and the logic behind it.

First the scanner finds all the folders that have an audio file (or an ebook file) and uses that folder as the book title (CD01 in your case).
Then it checks how many folders deep the book is to determine if the next folder down should be a series folder, author folder or is in the root directory. In your case, as you pointed out, BOOK was the series and SERIES would have been the author.

Here is why it lost those books though. The folder watcher detected a file was moved into the series folder (BOOK in your case) so it ignored the file. Any file moved into a non-book folder is ignored.
Then you deleted the book folder so the books were marked as not found.

The solution is to remove those books from Abs and scan the library again. There is no good way to resolve this once the scanner has read the CD folders as book folders other than creating a new folder in the SERIES directory and moving the files in.

The scanner could be improved to look for CD folders, but it would need to be fairly specific to prevent false positives. The folder watcher should probably have some sort of reporting for audio files in non-book folders so it can be spotted in the UI.

**Reply by snowmangoh** [reply]
Oh! So... If you delete those sub-folders, the series doesn't show up anymore because it is "empty", but the entry still exists in a table somewhere that there is a series folder there, right? What if there were a filter option under series to just show empty series, and deleting the SERIES from ABS removed that folder reference from that table? Then a normal scan would detect what it previously believed to be a series folder as a book folder again, right?

**Reply by advplyr** [reply]
It's not an empty series though, the series just has missing books in it. You can filter by Issues and batch delete missing books.

**Reply by snowmangoh** [reply]
Right, kinda... So I deleted those "books" (CD01, CD02, etc), but had already named the parent folder the correct book name (Book 01 - Book Name)... So the "series" I would be deleting would be the actual book name (Book 01 - Book Name) in the actual series. But the same problem exists for a couple of books that aren't actually in series too... it just thinks they were because there was a folder in the book folder... so the same solution could apply. All that really is needed is a way to say "forget you knew this folder was here" so it can detect it again without doing a forced rescan.
